content/lessons/chapter-4/public-key-4.tsx

```diff
@@ -4,122 +4,163 @@ import { ScriptingChallenge, LessonInfo, CodeExample } from 'ui'
 import { EditorConfig } from 'types'
 import { useTranslations } from 'hooks'
 import { Text } from 'ui'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 import { getLessonKey } from 'lib/progress'
+import { getData } from 'api/data'
 
 export const metadata = {
   title: 'chapter_four.public_key_four.title',
   key: 'CH4PKY4',
 }
-const javascript = {
-  program: `console.log("KILL")`,
-  defaultFunction: {
-    name: 'findHash',
-    args: ['nonce'],
-  },
-  defaultCode: `const crypto = require('crypto')
 
-// Create a program that finds a sha256 hash starting with 5 zeroes.
-// To submit your answer, log it to the terminal using console.log().
+const javascriptMatch = (match: string) => {
+  if (match === '\\"') {
+    return ''
+  } else if (match === ',' || match === '}') {
+    return 'n,'
+  }
+  return match // return the match unchanged if no replacements are made
+}
 
-// Type your code here
-`,
-  validate: async (answer) => {
-    if (!answer.startsWith('00000')) {
-      return [false, 'Hash must start with 5 zeroes.']
-    }
+export default function PublicKey4({ lang }) {
+  const [prevData, setPrevData] = useState({ lessonId: '', data: '' })
+  const newJavascriptData = prevData.data.replace(
+    /(\\"|[,\}])/g,
+    javascriptMatch
+  )
 
-    if (answer.length !== 64) {
-      return [false, 'Hash must be 64 characters long']
-    }
+  const newPythonData = prevData.data
+    .replace(/\\"[x]\\"/g, "'x'")
+    .replace(/\\"[y]\\"/g, "'y'")
+    .replace(/\\"/g, '')
```

Comment:
I think we can keep the object returned from the standard object data. I don't think we need this fancy footwork replacements